src/psyclone/tests/psyir/frontend/fparser2_select_case_test.py

```diff
@@ -95,6 +96,37 @@ def test_handling_case_construct():
     assert len(ifnode.else_body[0].children) == 2  # SELECT CASE ends here
 
 
+def test_very_large_select_construct_increases_recursion_limit(fortran_reader):
+    '''Test that a select case with a very large number of CASES preventively
+    increases the python recursion limit.'''
+
+    default_recursion_limit = sys.getrecursionlimit()
+
+    repetitions = 200
+    code = '''
+    subroutine test_subroutine(a)
+        use other, only: my_sub
+        integer, intent(in) :: a
+
+        SELECT CASE(a)
+    '''
+    for i in range(repetitions):
+        code += f'''
+        CASE ({i})
+            call my_sub()
+        '''
+    code += '''
+        END SELECT
+    end subroutine'''
+    _ = fortran_reader.psyir_from_source(code)
+
+    # Check that the recursion limit has been increased
+    assert sys.getrecursionlimit() > default_recursion_limit
+
+    # Restore default value
+    sys.setrecursionlimit(default_recursion_limit)
```

Comment:
How safe is this? Should we perhaps do the body of the test within a `try` and then have this in a `finally`?

Reply:
Alternatively, we could `monkeypatch` `sys.setrecursionlimit()` so that it just raises an error or something and check that the frontend causes it to be called?

Reply:
Done (with the try-finally approach)